src/api/internal.rs

```diff
@@ -433,6 +436,24 @@ impl<T: Pixel> ContextInner<T> {
     Ok(())
   }
 
+  #[allow(unused)]
+  pub fn build_dump_properties() -> PathBuf {
+    let mut out_dir = PathBuf::new();
+    let mut data_location = PathBuf::new();
+    out_dir.push(&env::var_os("CARGO_MANIFEST_DIR").unwrap());
+
+    if env::var_os("RAV1E_DATA_PATH").is_some() {
+      data_location.push(&env::var_os("RAV1E_DATA_PATH").unwrap());
+      fs::create_dir_all(data_location.clone()).unwrap();
+      data_location
+    } else {
+      data_location.push(&env::var_os("CARGO_MANIFEST_DIR").unwrap());
```

Comment:
Not sure if this is the best idea since:
- You have to use `cargo` to get the manifest dir environment variable (so just running the binary built with data dumping will panic out of the box). Idk if it's a big deal considering this is a debug feature anyway.
- The `target/out` folder might not make the most sense for this sort of output if we consider its usual subfolders:
  - `debug`, `release`
  - `<target triple>` for building with `--target=triple`
  - `rls` for RLS
  - `doc` for `cargo doc`

Reply:
Ah yes, 
and this feature is not enabled by default tho, anyone needs to compile with `--feature` flag to have this,

and yes, this is not a portable solution tbh, so I am modifying it to something like this

```diff
-      data_location.push(&env::var_os("CARGO_MANIFEST_DIR").unwrap());
-      data_location.push("target/out/");
+      data_location.push(&env::current_dir().unwrap());
+      data_location.push("out/");
       fs::create_dir_all(data_location.clone()).unwrap();
```

So it creates out/ folder where the data will be dumped, which seems to be more reasonable. 
Is that ok?

Reply:
Maybe something more telling instead of `out`?

Reply:
Not sure what to keep, one option would be getting filename from binary or something like that, but I think it maybe overkill for debug feature because,
+ This is only fall-back case when user do have env var,
+ Right now it is making out dir from $PWD.

On quick conversation in channel,

Chat Logs:

```irc
18:00 <lu_zero> this feature is a debug feature
18:00 <lu_zero> and you could restrict its scope as much as needed
18:00 <lu_zero> in the end you want to not clutter the root-directory
18:01 <lu_zero> so ideally you could just make it always output to .{descriptive-name}/
18:02 <lu_zero> and have an env-override in case somebody wants to explicitly give it a full path
18:02 <lu_zero> so you do not need to check for cargo-specific env
18:02 <lu_zero> or such
18:02 <mindfreeze> right now, we are not having any cargo-specific ones
18:03 <lu_zero> just by default write to ${PWD}/.some-name-you-decide
18:03 <mindfreeze> also updated the patchset with join and with_extension, please do check it :) 
```

Reply:
Now it is with `.lookahead_data` as discussed in IRC :)